ambari-server/src/main/java/org/apache/ambari/server/configuration/Configuration.java

```diff
@@ -2853,6 +2855,40 @@ private static Properties readConfigFile() {
     return properties;
   }
 
+  /**
+   * Removing the given properties from ambari.properties (i.e. at upgrade time)
+   *
+   * @param propertiesToBeCleared
+   *          the properties to be removed
+   * @throws AmbariException
+   *           if there was any issue when clearing ambari.properties
+   */
+  public void removePropertiesFromAmbariProperties(Collection<String> propertiesToBeRemoved) throws AmbariException {
+    try {
+      final File ambariPropertiesFile = new File(Configuration.class.getClassLoader().getResource(Configuration.CONFIG_FILE).getPath());
```

Comment:
You should use `org.apache.ambari.server.configuration.Configuration#readConfigFile` to read the properties from the `ambari.properties file` into a `java.util.Properties` object.   Then later write out the properties using `org.apache.ambari.server.configuration.Configuration#writeConfigFile` (which does not yet exist).